src/HealthChecks.Redis/DependencyInjection/RedisHealthCheckBuilderExtensions.cs

```diff
@@ -66,6 +66,37 @@ public static IHealthChecksBuilder AddRedis(
            timeout));
     }
 
+    /// <summary>
+    /// Add a health check for Redis services.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="connectionStringFactory">A factory to build the Redis connection string to use.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'redis' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddRedis(
+        this IHealthChecksBuilder builder,
+        Func<IServiceProvider, CancellationToken, Task<string?>> connectionStringFactory,

```

Comment:
This overload feels weird to me. It is too specific to the Aspire use case.

If you need to do something async, I think you should create the IConnectionMultiplexer asynchronously. Getting the connection string asynchronously isn't very common.

Reply:
I'm not sure if delayed acquisition of connection strings is Aspire specific. It certainly impacts Aspire, but there are plenty of times that the connection string is not immediately available (particularly in multi-tenant scenarios).